epub33/core/index.html

```diff
@@ -5984,10 +5984,119 @@ <h4>URLs in the OCF Abstract Container</h4>
+							</details>
+						</li>
+
+						<li>
+							Set <var>base</var> to be the <a data-cite="url#concept-base-url">base URL</a> that must be used to parse <var>url</var> as defined by the context (document or environment) where <var>url</var> is used, and according to the <a>content URL</a> of the <a>Package Document</a> (see <a href="#sec-parse-package-urls"></a>). 
+						</li>
+
+						<li>
+							Set <var>testURLRecord</var> to be the result of applying the <a data-cite="url#concept-url-parser">URL parser</a> to <var>url</var>, with <var>base</var>.
+						</li>
+
+						<li>
+							Let <var>testURLStringB</var> be the result of applying the <a data-cite="url#concept-url-serializer">URL Serializer</a> to <var>testURLRecord</var>.
+						</li>
+
+						<li>
+							If <var>testURLStringA</var> does not start with <code>https://a.example.org/</code> or <var>testURLStringB</var> does not start with <code>https://b.example.org/</code>, return <var>true</var>.
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">
+									If any of the result does not share the test URL host, it means that <var>url</var>, or
+									its base URL (for example, in HTML, if it is explicitly set with the <code>base</code> 
+									element), was <em>absolute</em> and points outside the container. This is acceptable.
+								</p>
+							</details>
+						</li>
+
+						<li>
+							If <var>testURLStringA</var> starts with <code>https://a.example.org/A/</code> and <var>testURLStringB</var> starts with <code>https://b.example.org/B/</code>, return <var>true</var>.
+							<details>
+								<summary>Explanation</summary>
+								<p class="note">The presence of the first test path segments (<code>A</code>, respectively <code>B</code>) indicate that the URL doesn't leak outside the container.</p>
+							</details>
+						</li>
+
+						<li>Return <var>false</var>.</li>
+					</ol>
+
+					<p>
+						In the <a>OCF Abstract Container</a>, any URL string MUST be an <a data-cite="url#absolute-url-with-fragment-string">absolute-URL-with-fragment-string</a> or a <a>valid-relative-container-URL-string</a>.
+					</p>
+
+					<div class="note">
+						<p>	
+							The properties of the <a>container root URL</a> are such that whatever the amount
+							of <a data-cite="url#double-dot-path-segment">double-dot path segments</a> in a URL string (for
+							example, <code>../../../secret</code>), it is parsed to a <a>content URL</a> (and not "leak"
+							outside the container). This avoids potential run-time security issues. 
+							The additional constraint on <a>valid-relative-container-URL-strings</a>  
+							ensures that such potentially problematic URLs can also be detected when checking the EPUB Document.
```

Comment:
EPUB Document looks like it should be EPUB Publication here.

This is exclusive of meta-inf files, correct, or does it handle the difference in where the paths are resolved relative to?

It might need to be "OCF Abstract Container" if it applies to all file types, as we don't have a word that. The files sit above a publication.

Reply:
I have changed for Publication.

Reply:
You are right about the META-INF files, though. I would not worry about it too much in the Note (that does not have to be 100% precise). But I have the impression that the algorithm may not be 100% precise either for META-INF: step (2) (and (6)) seems to work only for content document or package document URL-s. The META-INF directory is not mentioned... 

I will think about it tomorrow, it is too late now for me... (unless someone beats me into how to add that to the algorithm without making it horribly complicated...)

Reply:
Actually... I could argue that, in step 2/6

> Let base be the base URL that must be used to parse url as defined by the context (document or environment) where url is used

covers the META-INF case: the context  specifies that the base URL must be the container Root URL, per section 6.1.6.2. It is a bit subtle, and may be worth adding to explanation note. But normative text may just be fine as is...

Reply:
@iherman yes, exactly 👍 the algorithm does (intend to) cover the META-INF files.

As for EPUB Publication vs. OCF Abstract Container (as raised by @mattgarrish), I believe EPUB Publication is more readable in the note. The normative statement, however, does say "OCF Abstract Container", which is theoretically more encompassing. In short, we're all good 😊.

Reply:
> the algorithm does (intend to) cover the META-INF files

Okay, thanks for clarifying that. I had to run out when I posted this and I couldn't tell from the markup diff which was the case so wasn't sure exactly which to suggest. Publication is fine with me, as it is just a note after all.